app/gui/src/dashboard/hooks/backendBatchedHooks.ts

```diff
@@ -187,12 +189,49 @@ export function moveAssetsMutationOptions(backend: Backend) {
     mutationFn: async ([ids, parentId]: [ids: readonly AssetId[], parentId: DirectoryId]) => {
       const results = await Promise.allSettled(
         ids.map((id) =>
-          backend.updateAsset(id, { description: null, parentDirectoryId: parentId }, '(unknown)'),
+          backend
+            .updateAsset(id, { description: null, parentDirectoryId: parentId }, '(unknown)')
+            .catch((error) => {
+              if (error instanceof DuplicateAssetError) {
+                return { id, error }
+              }
+              throw error
+            }),
         ),
       )
+
+      const duplicateErrors = results
+        .filter((result) => result.status === 'fulfilled')
+        .map((result) =>
+          typeof result.value === 'object' && 'error' in result.value ? result.value : null,
+        )
+        .filter((error) => error != null)
+
       const errors = results.flatMap((result): unknown =>
         result.status === 'rejected' ? [result.reason] : [],
       )
+
+      if (duplicateErrors.length !== 0) {
+        const resolutions = await resolveDuplications({
+          targetId: parentId,
+          conflictingIds: duplicateErrors.map((error) => error.id),
+        })
+
+        // TODO: support replacements
+        const renames = resolutions.filter((resolution) => resolution.conclusion === 'rename')
+
+        await Promise.allSettled([
+          ...renames.map((resolution) =>
```

Comment:
`[...]`?!

Reply:
It’s a leftover from when I used to merge two resolution promises: `renames` and `replaces`. Now that we’ve disabled the `replace` option, it makes sense to remove the destruction associated with it. Thanks for pointing that out 🙂